read.go

```diff
@@ -128,13 +128,13 @@ func (r *reader) readTOCSections(toc *indexTOC, tags []string) error {
 				// If we don't recognize the section, we may be reading a newer index than the current version. Use
 				// a "dummy section" struct to skip over it.
 				skipSection = true
-				log.Printf("encountered malformed index section (%s), skipping over it", tag)
+				log.Printf("encountered unrecognized index section (%s), skipping over it", tag)
 
 				switch sectionKind(kind) {
 				case sectionKindSimple:
 					sec = &simpleSection{}
 				case sectionKindCompound:
-					sec = &lazyCompoundSection{}
+					sec = &compoundSection{}
```

Comment:
I went back to `compoundSection` for simplicity/ consistency. Now that we have `skip`, it's just as efficient to use `compoundSection`.